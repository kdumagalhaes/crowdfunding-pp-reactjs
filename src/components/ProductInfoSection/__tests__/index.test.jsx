import { fireEvent, render, screen } from "@testing-library/react";
import { ProductInfoSection } from "..";

describe("ProductInfoSection", () => {
    it("initial component should match snapshot", () => {
        const { asFragment } = render(<ProductInfoSection />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render the bookmark button icon", () => {
        render(<ProductInfoSection />);
        const bookmarkBtnImg = screen.getByAltText(/bookmark icon/i);
        expect(bookmarkBtnImg).toBeVisible();
    });

    it("should change the bookmark text and button when bookmark button is clicked", () => {
        render(<ProductInfoSection />);

        expect(screen.getByText("Bookmark")).toBeVisible();
        expect(screen.queryByText("Bookmarked")).not.toBeInTheDocument();
        expect(screen.getByAltText("bookmark icon")).toBeVisible();
        expect(
          screen.queryByAltText("bookmark icon active")
        ).not.toBeInTheDocument();

        const bookmarkBtn = screen.getByText("Bookmark");
        fireEvent.click(bookmarkBtn);

        expect(screen.queryByText("Bookmark")).not.toBeInTheDocument();
        expect(screen.getByText("Bookmarked")).toBeVisible();
        expect(screen.queryByAltText("bookmark icon")).not.toBeInTheDocument();
        expect(screen.getByAltText("bookmark icon active")).toBeVisible();
    });
});
