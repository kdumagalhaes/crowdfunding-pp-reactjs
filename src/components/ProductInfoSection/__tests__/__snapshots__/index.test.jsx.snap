// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfoSection initial component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL jnOboy"
  >
    <section
      class="sc-gsDKAQ egttvD"
    >
      <img
        alt="logo mastercraft"
        class="product-logo"
        src="logo-mastercraft.svg"
      />
      <h1
        class="product-name"
      >
        Mastercraft Bamboo Monitor Riser
      </h1>
      <p
        class="product-summary"
      >
        A beautiful & handcrafted monitor stand to reduce neck and eye strain.
      </p>
      <div
        class="flex-container"
      >
        <button
          class="sc-dkPtRN edUmWF"
          height="60px"
        >
          Back this project
        </button>
        <button
          class="btn-bookmark"
        >
          <img
            alt="bookmark icon"
            src="icon-bookmark.svg"
          />
          <p>
            Bookmark
          </p>
        </button>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
